### DISCUSSION TITLE
ModuleNotFoundError: No module named 'app.routes'

### DISCUSSION THREAD

**Discussion by michaelkremenetsky** [fastapi/fastapi#6523] [Questions] [answered]
So I'm learning fastapi right now and I was trying to separate my project into multiple files but when I do I get this error.

```ModuleNotFoundError: No module named 'app.routes'```

I have read [This][1] multiple times and I'm pretty sure I did everything right can anyone tell me what I did wrong?

```
app
│   main.py
│   __init__.py
│
└───routes
        auth.py
        __init__.py

```

main.py
```
from fastapi import FastAPI
from app.routes import auth

app = FastAPI()

app.include_router(auth.router)
```

auth.py
```
from fastapi import APIRouter

router = APIRouter()

@router.get("/test")
async def test():
	return {"test": "test"}
```

I tried both ```uvicorn main:app --reload``` and ```uvicorn app.main:app --reload```

Also I'm on Windows 11 using Python 3.9.6 and I've already made sure its in my path

  [1]: https://fastapi.tiangolo.com/tutorial/bigger-applications/

**Comment by panla** [comment, marked as answer]
two method to solve it

### first

- project
  - app
    - main.py
    - `__init__.py`
    - routes
      - auth.py
      - `__init__.py`
  - run.py

```bash
cd project
uvicorn app.main:app --reload
```

### second

- project
  - app
    - main.py
    - `__init__.py`
    - routes
      - auth.py
      - `__init__.py`
  - run.py

run.py

```python
from app.main import app
```

```bash
cd project
uvicorn run:app --reload
```

**Comment by michaelkremenetsky** [comment]
Thank you that fixed it 👍
